tests/test_smoke.py

```diff
@@ -133,7 +133,8 @@ def run_one_test(test: Test) -> Tuple[int, str, str]:
            f'\nLog: less {log_file.name}\n')
     test.echo(msg)
     log_file.write(msg)
-    if proc.returncode == 0 and test.teardown is not None:
+    if (proc.returncode == 0 or
+            pytest.terminate_on_failure) and test.teardown is not None:
```

Comment:
nit: Update https://github.com/skypilot-org/skypilot/blob/master/CONTRIBUTING.md#testing for this flag? 

Btw, I again on instinct looked at the top of this file for "Example usage:". I think we should copy the snippet in that markdown into top of this module.

Reply:
Added both. Thanks!